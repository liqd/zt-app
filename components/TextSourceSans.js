import React from 'react'
import { Text, StyleSheet } from 'react-native'
import { FONTS } from '../theme/fonts'

export const TextSourceSans = (props) => {
  const mergedStyles = props.style
    ? [styles.font, props.style]
    : styles.font
  return (
    <Text {...props} style={mergedStyles}>
      {props.children}
    </Text>
  )
}

const styles = StyleSheet.create({
  font: {
    fontFamily: FONTS.familySans,
  }
})
