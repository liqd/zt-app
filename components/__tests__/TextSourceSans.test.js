import React from 'react';
import { TextSourceSans } from '../TextSourceSans.js';
import { render } from '@testing-library/react-native';

test('Test TextsourceSans no style in props', () => {
  const { toJSON } = render(<TextSourceSans>{'TestText'}</TextSourceSans>);
  expect(toJSON()).toMatchSnapshot();
});

test('Test TextsourceSans no style in props', () => {
  const { toJSON } = render(
    // eslint-disable-next-line react-native/no-inline-styles
    <TextSourceSans style={{ fontSize: 10 }}>{'TestText'}</TextSourceSans>
  );
  expect(toJSON()).toMatchSnapshot();
});
