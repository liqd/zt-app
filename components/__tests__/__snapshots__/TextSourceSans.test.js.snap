// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test TextsourceSans no style in props 1`] = `
<Text
  style={
    Object {
      "fontFamily": "SourceSansPro_400Regular",
      "fontSize": 18,
    }
  }
>
  TestText
</Text>
`;

exports[`Test TextsourceSans no style in props 2`] = `
<Text
  style={
    Array [
      Object {
        "fontFamily": "SourceSansPro_400Regular",
        "fontSize": 18,
      },
      Object {
        "fontSize": 10,
      },
    ]
  }
>
  TestText
</Text>
`;
